src/sessionid.ts

```diff
@@ -26,6 +28,10 @@ export class SessionIdManager {
         this._sessionStartTimestamp = null
         this._sessionActivityTimestamp = null
 
+        if (this.config.on_session_id_changed_fn) {
```

Comment:
Optional API design idea:

This seems a bit awkward to use to me, since I have to copy all my business logic to posthog at initialisation time, when it might not exist (say, I need to load user data so I can feed these new sessionID values to the backend, bla bla). I also can't change this during the session posthog is initialised.

Another option here is to expose the callback function instead: Whenever the sessionID changes, we call the onSessionIdChanged with whatever has happened. You can remove your callback fns, add new ones, attach in the middle when you're ready to handle, etc. etc.

See `onFeatureFlags()` callback for an example of this.

Reply:
That's _way_ nicer 💖